RuboCop upgrade

natesholland opened the pull request:
I am trying to get more involved in open source code maintenance. In order to do this I figured I would try upgrading RuboCop for a few gems I use frequently to help them stay up to date. RuboCop on this project was very out of date so I started by upgrading it by ten minor versions to `0.59.2` but I can go further if the maintainers want.

I broke this PR into different commits based on what work I was doing. The first commit is the update and auto-fixes. The commits after that are the commits where I made judgement calls on what to do.

Please let me know if there's anything I can do to help or improve.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/httprb/http/548)
<!-- Reviewable:end -->

mikegee commented on spec/support/black_hole.rb:
```diff
@@ -2,9 +2,11 @@
 
 module BlackHole
   class << self
+    # rubocop:disable Style/MethodMissingSuper
```
You should be able to move this down as a suffix of the `def` and remove the `enable` comment afterwards:

```ruby
def method_missing(*) # rubocop:disable Style/MethodMissingSuper
```

mikegee commented on lib/http/options.rb:
```diff
@@ -196,3 +196,4 @@ def argument_error!(message)
     end
   end
 end
+# rubocop:enable Metrics/ClassLength
```
Maybe you can move the `disable` comment down to the line that opens the class and we won't need this? For instance:

```ruby
module HTTP
  class Options # rubocop:disable Metrics/ClassLength
```

natesholland commented:
Thanks for the pointer @mikegee, I've made the recommended changes.

mikegee commented:
Awesome!

I didn't comment on `spec/support/dummy_server/servlet.rb` earlier, but I think the same recommendation applies there too.

natesholland commented:
@mikegee addressed.

As a side note, I would like to get #546 merged before mine if possible. I really want that change in place and don't want my RuboCop changes to conflict with it.

natesholland commented:
@ixti I am new to the Reviewable tool but I believe I have addressed all of your feedback. If you want me to revert out the Style/EvalWithLocation changes in fb6d66e please let me know and I'll do that.

ixti commented:
@natesholland Honestly I just wanted to try our reviewable ;))

Re Style/EvalWithLocation - I'm totally fine with the requirement of file/line. I don't like `+ 1` part only. Although I see where does it comes from, so I guess let's keep it as you did it.

ixti commented:
> As a side note, I would like to get #546 merged before mine if possible. I really want that change in place and don't want my RuboCop changes to conflict with it.

That PR needs more work to be done. I will get to it as soon as I will have time for that. Thus merging this prior that change.

natesholland commented:
@ixti if I have time, would you like me to upgrade RuboCop again in the future? The version I bumped to now is much more recent, but still a ways behind the latest version.

ixti commented:
@natesholland Sure, feel free to do this. I'm all for keeping rubocop to the latest.
